lib/coap/client.ex

```diff
@@ -1,141 +1,162 @@
 defmodule CoAP.Client do
   @moduledoc """
-    CoAP Client interface
+  CoAP Client interface
   """
   alias CoAP.Message
-
-  import Logger, only: [debug: 1]
-
+  alias CoAP.Transport
+
+  # Default timeout: 5 sec
+  @default_timeout 5_000
+  @message_opts [:ack_timeout, :max_retransmit, :socket_adapter, :socket_opts]
+
+  @type option() ::
```

Comment:
👏

Reply:
I like the use of options.  Need to update the readme.

Reply:
updated